tests/unit/test_mips.py

```diff
@@ -1,121 +1,236 @@
+def test_get_secrets(ssm_stub):
+    '''
+    Test getting secrets from SSM
 
-        # collect secure parameters
+    First catch an exception if no secrets are found,
+    then assert correct processing of mock secrets.
+
+    '''
+
+    # create object under test
+    mips_app = mips_api.mips.App()
+
+    # no secrets in ssm
+    ssm_stub.add_client_error('get_parameters_by_path', service_error_code='ParameterNotFound')
+    with pytest.raises(Exception):
         mips_app.collect_secrets()
-        assert mips_app._ssm_secrets == ssm_secrets
-        stub_ssm.assert_no_pending_responses()
-
-        # get chart of accouts from mips
-        mips_app.get_mips_data(request_path)
-        assert mips_app.mips_dict == expected_mips_dict
-        assert login_mock.call_count == 1
-        assert chart_mock.call_count == 1
-        assert logout_mock.call_count == 1
-
-        # exception getting chart of accounts
-        requests_mock.get(mips_api.mips.App._mips_url_chart, exc=Exception)
-        with pytest.raises(Exception):
-            mips_app._collect_mips_data()
-        assert logout_mock.call_count == 2
-
-        # secrets are invalid
-        mips_app._ssm_secrets = {'foo': 'bar'}
-        with pytest.raises(Exception):
-            mips_app.collect_secrets()
-
-        # no secrets in ssm
-        mips_app._ssm_secrets = None
-        stub_ssm.add_client_error('get_parameters_by_path', service_error_code='ParameterNotFound')
-        with pytest.raises(Exception):
-            mips_app.collect_secrets()
+
+    # process secrets from ssm
+    ssm_stub.add_response('get_parameters_by_path', mock_ssm_params)
+    mips_app.collect_secrets()
+
+    # assert secrets were collected
+    assert mips_app._ssm_secrets == mock_secrets
+    ssm_stub.assert_no_pending_responses()
+
+
+@pytest.mark.order(after='test_mips_init')
+def test_get_data(requests_mock):
+    '''
+    Test getting data from upstream api
+
+    Relies on `requests-mock.Mocker` fixture to inject mock responses into `requests`.
+    '''
+
+    # inject mock responses into `requests`
+    login_mock = requests_mock.post(mips_api.mips.App._mips_url_login, json=mock_token)
+    chart_mock = requests_mock.get(mips_api.mips.App._mips_url_chart, json=mock_chart)
+    logout_mock = requests_mock.post(mips_api.mips.App._mips_url_logout)
+
+    # create object under test
+    mips_app = mips_api.mips.App()
+
+    # inject secrets
+    mips_app._ssm_secrets = mock_secrets
+
+    # get chart of accounts from mips
+    mips_app._collect_mips_data()
+    assert mips_app.mips_dict == expected_mips_dict
+    assert login_mock.call_count == 1
+    assert chart_mock.call_count == 1
+    assert logout_mock.call_count == 1
+
+    # assert logout is called when an exception is raised
+    requests_mock.get(mips_api.mips.App._mips_url_chart, exc=Exception)
+    with pytest.raises(Exception):
+        mips_app._collect_mips_data()
+    assert logout_mock.call_count == 2
```

Comment:
no, that wouldn't be thread safe if both tests could run simultaneously because we couldn't be sure about which function call would get which mock value. since we need to run the tests sequentially, why not run them in the same function?

Reply:
ok, i didn't realize that was the reason for running it sequentially.  can we add a comment about that somewhere?

Reply:
ok, i didn't realize that was the reason for making tests running sequentially.  can we comment about that somewhere?

Reply:
done